### PR TITLE
Add Arxiv classification dataset

### PR DESCRIPTION
**Pull request by shreeya-dhakal**
- [X] I have tested that the dataset runs with the `mteb` package.
- [X] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [X] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [X] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [X] I have considered the size of the dataset and reduced it if it is too big (2048 examples is typically large enough for most tasks)
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [X] Run tests locally to make sure nothing is broken using `make test`. 
- [X] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

File: mteb/tasks/Classification/eng/ArxivClassification.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        domains=None,
+        task_subtypes=None,
+        license=None,
+        socioeconomic_status=None,
+        annotations_creators=None,
+        dialect=None,
+        text_creation=None,
+        bibtex_citation=None,
+        n_samples={"test": 2500},
+        avg_character_length=None,
```
Comment (isaac-chung) [inline mteb/tasks/Classification/eng/ArxivClassification.py:33]: Please fill these out using the dataset page, the paper, and `mteb/abstasks/TaskMetadata.py` as references. e.g.
- license = "Not specified"
- bibtex = copy from dataset page
- dialect=[]
- domain=["Academic"]
- ...

**Review by isaac-chung — COMMENTED** [review]
Thanks for your contribution! Just a few comments.
Feel free to add 2 points for this dataset and 2 points for me for the review.

File: mteb/tasks/Classification/eng/ArxivClassification.py (hunk 1 above)
Comment (shreeya-dhakal) [reply mteb/tasks/Classification/eng/ArxivClassification.py]: I made the suggested changes!

File: mteb/tasks/Classification/eng/ArxivClassification.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            "revision": "f9bd92144ed76200d6eb3ce73a8bd4eba9ffdc85",
+        },
+        reference="https://ieeexplore.ieee.org/document/8675939",
+        type="Classification",
+        category="s2s",
+        eval_splits=["test"],
+        eval_langs=["eng-Latn"],
+        main_score="accuracy",
+        date=None,
+        form=None,
```
Comment (isaac-chung) [inline mteb/tasks/Classification/eng/ArxivClassification.py:23]:
```suggestion
        form=["written"],
```

File: mteb/tasks/Classification/eng/ArxivClassification.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            "path": "ccdv/arxiv-classification",
+            "revision": "f9bd92144ed76200d6eb3ce73a8bd4eba9ffdc85",
+        },
+        reference="https://ieeexplore.ieee.org/document/8675939",
+        type="Classification",
+        category="s2s",
+        eval_splits=["test"],
+        eval_langs=["eng-Latn"],
+        main_score="accuracy",
+        date=None,
```
Comment (isaac-chung) [inline mteb/tasks/Classification/eng/ArxivClassification.py:22]: A rough estimate will do: (yyyy-mm-dd, yyyy-mm-dd)

File: mteb/tasks/Classification/eng/ArxivClassification.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        form=None,
+        domains=["Academic"],
+        task_subtypes=None,
```
Comment (isaac-chung) [inline mteb/tasks/Classification/eng/ArxivClassification.py:25]:
```suggestion
        task_subtypes=["Topic classification"],
```

File: mteb/tasks/Classification/eng/ArxivClassification.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        license="Not specified",
+        socioeconomic_status=None,
```
Comment (isaac-chung) [inline mteb/tasks/Classification/eng/ArxivClassification.py:27]:
```suggestion
        socioeconomic_status="high",
```

File: mteb/tasks/Classification/eng/ArxivClassification.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        annotations_creators=None,
```
Comment (isaac-chung) [inline mteb/tasks/Classification/eng/ArxivClassification.py:28]:
```suggestion
        annotations_creators="derived",
```

File: mteb/tasks/Classification/eng/ArxivClassification.py [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+        dialect=[],
+        text_creation=None,
```
Comment (isaac-chung) [inline mteb/tasks/Classification/eng/ArxivClassification.py:30]:
```suggestion
        text_creation="found",
```

**Review by isaac-chung — COMMENTED** [review]
Thanks! Just a few small comments.

File: mteb/tasks/Classification/eng/ArxivClassification.py (hunk 3 above)
Comment (shreeya-dhakal) [reply mteb/tasks/Classification/eng/ArxivClassification.py:22]: Is this the date this dataset was compiled or the dates of the papers included in this dataset?

Comment (isaac-chung) [reply mteb/tasks/Classification/eng/ArxivClassification.py]: Compiled. But what you have is fine.

**Review by isaac-chung — APPROVED** [review]
Thanks!

**Pull request merged**